src/index.ts

```diff
@@ -137,6 +137,11 @@ export class Sema {
 			capacity?: number;
 		} = {}
 	) {
+		if ((nr | 0) !== nr) {
```

Comment:
```suggestion
		if (nr && !Number.isInteger(nr)) {
```

Reply:
actually, nr should be a positive non-zero integer, so how about `if (nr < 1 || (nr | 0) !== nr)`?

Reply:
Good point, that seems good to me!

Oh no, I didn't thought about that. The polyfill is actually quite simple:

```js
Number.isInteger = Number.isInteger || function(value) {
  return typeof value === 'number' && 
    isFinite(value) && 
    Math.floor(value) === value;
};
```

Of course we shouldn't actually polyfill it but maybe just do these checks in the if clause.

`Number.isFinite()` seems to have an implemented in all browsers.